code/modules/assembly/flash.dm

```diff
@@ -185,21 +191,33 @@
   *
   *	Returns the amount of 'deviation', 0 being facing eachother, 1 being sideways, 2 being facing away from eachother.
   * Arguments:
-  * * M - Victim
-  * * user - Attacker
+  * * victim - Victim
+  * * attacker - Attacker
   */
-/obj/item/assembly/flash/proc/calculate_deviation(mob/living/carbon/M, mob/user)
-	var/dir1 = M.dir
-	var/dir2 = turn(user.dir,180)
-	//Imagine 2 vectors coming from both mobs, they represent the direction the mob is currently looking towards,
-	//What we actually check is we check if the inverted vector of the second mob is equal to the vector of the first one which indicates they they are looking in the same line
-	//The second check makes sure that the first user is actually facing the mob, since the first check will fail at it's job when the 2 mobs face away from each other.
-	if(dir1 == dir2 && get_dir(M,user) == dir1)
-		return 0
-	else if((turn(dir1,90) == dir2 || turn(dir1,-90) == dir2) || M.loc == user.loc)
-		return 1
-	else if(turn(dir1,180) == dir2)
-		return 2
+/obj/item/assembly/flash/proc/calculate_deviation(mob/victim, atom/attacker)
+	// If the victim was looking at the attacker, this is the direction they'd have to be facing.
+	var/victim_to_attacker = get_dir(victim, attacker)
+	var/victim_dir = victim.dir
+
+	// Are they on the same tile? We'll return partial deviation. This may be someone flashing while lying down
+	// or flashing someone they're stood on the same turf as, or a borg flashing someone buckled to them.
+	if(victim.loc == attacker.loc)
+		return DEVIATION_PARTIAL
+
+	// Is the victim looking directly at the attacker? and is the attacker looking at the victim?
+	if(victim_dir == victim_to_attacker)
+		return DEVIATION_NONE
+
+	// Is the victim looking at the attacker within 45 degrees?
+	if(victim_dir == turn(victim_to_attacker, 45) || victim_dir == turn(victim_to_attacker, -45))
+		return DEVIATION_NONE
+
+	// Is the is the victim looking perpendicular to the attacker?
+	if(victim_dir == turn(victim_to_attacker, 90) || victim_dir == turn(victim_to_attacker, -90))
+		return DEVIATION_PARTIAL
+
+	// This only leaves directions from directly behind and diagonal-behind. No flash bueno.
+	return DEVIATION_FULL
```

Comment:
```suggestion
/obj/item/assembly/flash/proc/calculate_deviation(mob/victim, atom/attacker)

	// Are they on the same tile? We'll return partial deviation. This may be someone flashing while lying down
	// or flashing someone they're stood on the same turf as, or a borg flashing someone buckled to them.
	if(victim.loc == attacker.loc)
		return DEVIATION_PARTIAL

	// If the victim was looking at the attacker, this is the direction they'd have to be facing.
	var/victim_to_attacker = get_dir(victim, attacker)
	// The victim's dir is necessarily a cardinal value.
	var/victim_dir = victim.dir
	
	// Attacker within 45 degrees of where the victim is facing.
	if(victim_dir & victim_to_attacker)
		return DEVIATION_NONE

	// Attacker at 135 or more degrees of where the victim is facing.
	if(victim_dir & REVERSE_DIR(victim_to_attacker))
		return DEVIATION_FULL

	// Attacker lateral to the victim.
	return DEVIATION_PARTIAL
```
Might want to add comments of your own.

Reply:
Implemented with additional visual comments.